compat/src/suspense.js

```diff
@@ -215,35 +141,15 @@ Suspense.prototype.componentWillUnmount = function() {
  * @param {import('./internal').SuspenseState} state
  */
 Suspense.prototype.render = function(props, state) {
-	if (this._detachOnNextRender) {
-		// @TODO Is this logic still needed with the move to backing tree?
-		// When the Suspense's _vnode was created by a call to createVNode
-		// (i.e. due to a setState further up in the tree)
-		// it's _children prop is null, in this case we "forget" about the parked vnodes to detach
-		if (this._internal._children) {
-			// @TODO: Consider rebuilding suspense detached parent logic to use root nodes
-			const detachedParent = document.createElement('div');
-			const detachedComponent = this._internal._children[0]._component;
-			this._internal._children[0] = detachedClone(
-				this._detachOnNextRender,
-				detachedParent,
-				(detachedComponent._originalParentDom = detachedComponent._parentDom)
-			);
-		}
-
-		this._detachOnNextRender = null;
+	if (this._parentDom == null) {
+		this._parentDom = getParentDom(this._internal);
 	}
```

Comment:
`this._parentDom` is null on renders that aren't suspended. To ensure that Suspense behaves like a Fragment, we need to set the Portal's parentDOM prop to match Suspense's parentDOM. So we manually search for it here.

Reply:
Note, this is necessary because root nodes must have a non-null `_parentDOM` prop on initial render so they get the `TYPE_ROOT` flag. If instead we were to use another signal `e.g. vnode.type == Portal` or something then we could say "for root nodes with a nullish _parentDOM, default to the current parentDOM" in diff/components.js